src/Http/Livewire/LivewireDatatable.php

```diff
@@ -627,45 +649,90 @@ public function refreshLivewireDatatable()
         $this->page = 1;
     }
 
+    public function getColumnDirection(string $sortString): string
+    {
+        $direction = 'desc';
+        if (Str::contains($sortString, '|')) {
+            $direction = Str::after($sortString, '|');
+            if ($direction !== 'asc' && $direction !== 'desc') {
+                throw new \Exception("Invalid direction $direction given in sanitizeColumnSort() method. Allowed values: asc, desc.");
+            }
+        }
+
+        return $direction;
+    }
+
     /**
      * Order the table by a given column index starting from 0.
      *
-     * @param  int  $index  which column to sort by
-     * @param  string|null  $direction  needs to be 'asc' or 'desc'. set to null to toggle the current direction.
+     * @param int $index which column to sort by
+     * @param string|null $direction needs to be 'asc' or 'desc'. set to null to toggle the current direction.
      * @return void
      */
     public function sort($index, $direction = null)
     {
-        if (! in_array($direction, [null, 'asc', 'desc'])) {
+        if (!in_array($direction, self::ORDER_BY_DIRECTION_STATES)) {
```

Comment:
```suggestion
        if (! isset(self::ORDER_BY_DIRECTION_STATES[$direction])) {
```

`isset` is a lot faster than `in_array`

Reply:
$direction is the value here, like asc, desc not the array key. I cannot use isset in this case.